### PR TITLE
Simplify `hasBlankLine`

### PR DESCRIPTION
**Pull request by knz**
Found while investigating #14.

This really deserves a unit test, which is why I'd like to merge #14 or something equivalent (in fact I have added the unit test for this PR there).

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/datadriven/17)

### CODE REVIEW

File: datadriven.go [hunk 1]
```diff
@@ -496,12 +495,10 @@ func (td TestData) Fatalf(tb testing.TB, format string, args ...interface{}) {
 	tb.Fatalf("%s: %s", td.Pos, fmt.Sprintf(format, args...))
 }
 
+var blankLineRe = regexp.MustCompile(`(?m)^[\t ]*\n`)
```
Comment (RaduBerinde) [inline datadriven.go]: we can use the more typical $ instead of \n no?

Comment (knz) [reply datadriven.go]: Unfortunately, no (that was my first attempt).
Go does something very weird at the end of the input. I added a comment to explain.

**Comment by knz** [comment]
TFYR

File: datadriven.go [hunk 2]
```diff
@@ -496,12 +495,23 @@ func (td TestData) Fatalf(tb testing.TB, format string, args ...interface{}) {
...
-	for scanner.Scan() {
-		if strings.TrimSpace(scanner.Text()) == "" {
-			return true
-		}
-	}
-	return false
+	return blankLineRe.MatchString(s)
 }
+
+// blankLineRe matches lines that contain only whitespaces (or
```
Comment (RaduBerinde) [inline datadriven.go]: Nice comment, thanks!

**Pull request merged**